qir/qat/ValidationPass/ValidationPassConfiguration.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+#include "QatTypes/QatTypes.hpp"
+#include "ValidationPass/ValidationPassConfiguration.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    void ValidationPassConfiguration::setup(ConfigurationManager& config)
+    {
+        config.setSectionName("Validation configuration", "");
+        config.addParameter(
+            allow_internal_calls_, "allow-internal-calls", "Whether or not internal calls are allowed.");
+        config.addParameter(
+            save_report_to_, "save-validation-report", "Saves the validation report to specified filename.");
+    }
+
+    String ValidationPassConfiguration::profileName() const
+    {
+        return profile_name_;
+    }
+
+    ValidationPassConfiguration ValidationPassConfiguration::fromProfileName(String const& name)
+    {
+
+        auto profile = ValidationPassConfiguration();
+        if (name == "generic")
```

Comment:
Is this generic profile only for testing purposes? Or is there a production scenario associated with it?

Reply:
At the moment we do not have a production scenario other than you might want to bypass validation and this can be done by choosing a generic profile that allows everything. Later on we may want to add such things a name enforcing which is also done for the generic case.

Reply:
Oh, I misunderstood this profile and thought it was a deny all. Allow all makes sense as something we'd need for testing and possible future user scenarios. Thanks for explaining!